iceoryx2/tests/node_tests.rs

```diff
@@ -62,6 +62,10 @@ mod node {
             let view = match node {
                 NodeState::<S>::Alive(ref view) => view as &dyn NodeView,
                 NodeState::<S>::Dead(ref view) => view as &dyn NodeView,
+                NodeState::<S>::Inaccessible(_) | NodeState::<S>::Undefined(_) => {
+                    assert_that!(true, eq false);
+                    panic!();
```

Comment:
Why not just
```suggestion
                    panic!("NodeState should not be Inaccessible or Undefined");
```
Alternatively, instead of `panic`, an `unreachable` would semantically also make sense